### PR TITLE
Bug 1170151 - Migrate integration tests between SMS and Contacts to Gij

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e73ee184c710d153d8815139e6724c90df6b1355

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5f2fdb1438269cdcf63c439420cf4ec05f1e744e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0ee9c7bb59fe340c80a2560dc2e49bf489cf95c0

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=68865758668c9201614f0f1f481e1774bfc11933

File: apps/sms/test/marionette/lib/views/conversation/accessors.js [hunk 1]
```diff
@@ -9,7 +9,11 @@ var SELECTORS = Object.freeze({
   container: '#messages-container',
   callButton: '#messages-call-number-button',
   // TODO: Remove the next selector once Message has its own accessors
-  messageContent: '.message-content > p:first-child'
+  messageContent: '.message-content > p:first-child',
+  createNewContact: 'form.contact-prompt [data-l10n-id="createNewContact"]',
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: nit: I don't think we need ```form``` in the selector + please rename property to smth like ```createNewContactMenuItem``` or ```createNewContactMenuOption``` so that it's clear what it exactly is.

File: apps/sms/test/marionette/lib/views/conversation/accessors.js [hunk 2]
```diff
@@ -9,7 +9,11 @@ var SELECTORS = Object.freeze({
...
+  addToExistingContact: 'form.contact-prompt ' +
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: nit: same notes here

File: apps/sms/test/marionette/lib/views/conversation/accessors.js [hunk 3]
```diff
@@ -9,7 +9,11 @@ var SELECTORS = Object.freeze({
...
+    '[data-l10n-id="addToExistingContact"]',
+  subHeaderPhoneNumber: '#contact-carrier .phone-number'
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: nit: I'd say we usually name it as ```carrierHeader``` rather than ```subHeader```

File: apps/sms/test/marionette/lib/views/conversation/view.js [hunk 4]
```diff
@@ -7,13 +7,23 @@ var ConversationAccessor = require('./accessors');
...
 var requireFromApp = require(appRoot +
-  '/shared/test/integration/require_from_app').requireFromApp;
+  '/shared/test/integration/require_from_app').require;
 
 function ConversationView(client) {
   this.client = client;
   this.accessors = new ConversationAccessor(client);
 }
 
+ConversationView.prototype = {
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/view.js]: Well, let's use either ```ConversationView.prototype = {``` or ```ConversationView.prototype.callContact = ``` notation, but not both. I'd prefer first, but would not object against second.

File: apps/sms/test/marionette/lib/views/conversation/view.js [hunk 5]
```diff
@@ -7,13 +7,23 @@ var ConversationAccessor = require('./accessors');
...
+  get headerTitle() {
+    return this.accessors.headerTitle.text();
+  },
+
+  get subHeaderPhoneNumber() {
```
Comment (azasypkin) [inline apps/sms/test/marionette/lib/views/conversation/view.js]: nit: same note about ```carrierHeader``` vs ```subHeader```

File: tests/integration/sms/contacts_interaction_test.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('chai').assert;
+
+var appRoot = require('app-root-path');
+// TODO Change the path once requireFromApp becomes its own module
+var FromApp = require(appRoot + '/shared/test/integration/require_from_app');
+
+var Messages = FromApp.require('sms', 'lib/messages');
```
Comment (azasypkin) [inline tests/integration/sms/contacts_interaction_test.js]: not related to this patch, just feedback from the user of your lib :) I'd rather have ```fromApp('sms').require('lib/messages')```;

File: tests/integration/sms/contacts_interaction_test.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+marionette('Contacts', function() {
+  var messagesApp;
+
+  var client = marionette.client({
+    prefs: {
+      'focusmanager.testmode': true
+    }
+  });
+
+  var thread;
```
Comment (azasypkin) [inline tests/integration/sms/contacts_interaction_test.js]: since this file is out of scope of my responsibility, here just feedback you can ignore :)

We usually join uninitialized variables together ```var a, b, c;```, but never do this for initialized variables.

File: tests/integration/sms/contacts_interaction_test.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  };
+
+  setup(function() {
+    var contactsApp = new Contacts(client);
+    contactsApp.launch();
+    contactsData = new ContactsData(client);
+    contactsData.createMozContact(testContact);
+    // Closing the application is needed, in order to be sure to switch to the
+    // right frame when an activity is started
+    contactsApp.close();
```
Comment (azasypkin) [inline tests/integration/sms/contacts_interaction_test.js]: You can add reference to https://bugzilla.mozilla.org/show_bug.cgi?id=1140344 that was trying to get rid of this nasty behavior, but was totally ignored by reviewer :)

File: apps/sms/test/marionette/lib/views/conversation/accessors.js (hunk 1 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: Done

File: apps/sms/test/marionette/lib/views/conversation/accessors.js (hunk 2 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: Done

File: apps/sms/test/marionette/lib/views/conversation/accessors.js (hunk 3 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/lib/views/conversation/accessors.js]: Done

File: apps/sms/test/marionette/lib/views/conversation/view.js (hunk 4 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/lib/views/conversation/view.js]: Done, the first way.

File: apps/sms/test/marionette/lib/views/conversation/view.js (hunk 5 above)
Comment (JohanLorenzo) [inline apps/sms/test/marionette/lib/views/conversation/view.js]: Done.

File: tests/integration/sms/contacts_interaction_test.js (hunk 6 above)
Comment (JohanLorenzo) [inline tests/integration/sms/contacts_interaction_test.js]: You're right, that's way better! I changed it.

File: tests/integration/sms/contacts_interaction_test.js (hunk 7 above)
Comment (JohanLorenzo) [inline tests/integration/sms/contacts_interaction_test.js]: Done.

File: tests/integration/sms/contacts_interaction_test.js (hunk 8 above)
Comment (JohanLorenzo) [inline tests/integration/sms/contacts_interaction_test.js]: Done.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2c32f1f3baeea11f428ae4f4388d3e1a1a6aa63e

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1fe8bc6266f2873798fc803cbea33b2dbc44f9b7
